lmdk/modules/smart_amp_test/smart_amp_test.c

```diff
@@ -0,0 +1,317 @@
+// SPDX-License-Identifier: BSD-3-Clause
+//
+// Copyright(c) 2020 Intel Corporation. All rights reserved.
+//
+// Author: Bartosz Kokoszko <bartoszx.kokoszko@linux.intel.com>
+// Author: Andrula Song <andrula.song@intel.com>
+// Author: Chao Song <chao.song@linux.intel.com>
+
+#define MODULE_BUILD
+
+#include <string.h>
+#include <stdint.h>
+#include <stddef.h>
+
+#include <module/base.h>
+#include <module/api_ver.h>
+#include <module/interface.h>
+#include <iadk/adsp_error_code.h>
+#include <rimage/sof/user/manifest.h>
+#include <audio/source_api.h>
+#include <audio/sink_api.h>
+#include "smart_amp_test.h"
+#include <ipc4/module.h>
+
+static struct smart_amp_data smart_amp_priv;
+
+static int smart_amp_init(struct processing_module *mod)
+{
+	const struct ipc4_base_module_extended_cfg *base_cfg;
+	struct module_data *mod_data = &mod->priv;
+	struct smart_amp_data *sad;
+	uint8_t *pin_data;
+	int bs;
+	int ret;
+
+	base_cfg = mod_data->cfg.init_data;
+	sad = &smart_amp_priv;
```

Comment:
can and shall we add a check that this function isn't called when 1 instance of the module is already running?

Reply:
Only one smart amp in topology and the pcm can't be used two times at the same time, so no need to check it.